### DISCUSSION TITLE
Has anyone experienced/noticed upstream cluster stat reset during istio upgrades?

### DISCUSSION THREAD

**Discussion by qudongfang** [istio/istio#48379] [General]
:raised_hands: Has anyone experienced/noticed upstream cluster stat reset during istio upgrades? we've seen this when upgrading istio from 1.17 to 1.18 and from 1.18 to 1.19 again

symptoms:

- stat counters of the upstream clusters on the service consumer istio-proxy side get reset, such as `istio_requests_total{reporter="source"}`
- service consumer istio-proxy re-connects to the service, which causes a “thundering herd” 

I wonder if something that might cause cluster rebuild happened when the istio-proxy gets re-connected to a newer version of istiod(pilot)

Any help would be greatly appreciated

**Comment by howardjohn** [comment]
It's pretty likely related to what you said about new xds config resetting things 

If you use the canary upgrade approach this entire class of issues goes away, as a proxy never switches the version of the control plane it uses
